pdnshttp/client.go

```diff
@@ -19,8 +20,17 @@ type Client struct {
 
 // NewClient returns a new PowerDNS HTTP client
 func NewClient(baseURL string, hc *http.Client, auth ClientAuthenticator, debugOutput io.Writer) *Client {
+	u, err := url.ParseRequestURI(baseURL)
+	if err != nil {
+		panic(err)
+	}
+	if strings.TrimSuffix(u.Path, "/") == "" {
+		u.Path = "/api/v1"
+	} else {
+		u.Path = strings.TrimSuffix(u.Path, "/")
+	}
 	c := Client{
-		baseURL:       baseURL,
+		baseURL:       u.Scheme + "://" + u.Host + u.Path,
```

Comment:
[`URL.String()`](https://pkg.go.dev/net/url#URL.String) should do exactly this, and also cover most of the other edge cases that might be in an URL (username+password, for example).

```suggestion
		baseURL:       u.String()
```

Reply:
Ah didn't know. I'm not familiar with go at all :)